WalletWasabi/Services/SystemAwakeChecker.cs

```diff
@@ -11,21 +11,36 @@
 
 namespace WalletWasabi.Services
 {
-	public class SystemAwakeChecker : PeriodicRunner
+	public class SystemAwakeChecker
 	{
 		private const string Reason = "CoinJoin is in progress.";
 		private static readonly TimeSpan Timeout = TimeSpan.FromMinutes(5);
 
+		private bool _canShutdown = true;
+		private object _shutDownLock = new();
+
 		private volatile IPowerSavingInhibitorTask? _powerSavingTask;
 
-		private SystemAwakeChecker(WalletManager walletManager, Func<Task<IPowerSavingInhibitorTask>>? taskFactory) : base(TimeSpan.FromMinutes(1))
+		private SystemAwakeChecker(WalletManager walletManager, Func<Task<IPowerSavingInhibitorTask>>? taskFactory)
 		{
 			WalletManager = walletManager;
 			TaskFactory = taskFactory;
 		}
 
+		public bool CanShutdown
+		{
+			get => _canShutdown;
```

Comment:
You must use `lock (_shutDownLock)` here too.

btw: I recommend to read https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/lock-statement#example

Reply:
Didn't know that. Thanks!